### PR TITLE
Fix spacing in Markdown lists and definitions

### LINKED ISSUE #9: Bug: Modified spacing for Markdown lists in the Changelog
TL;DR: This action adds two extra spaces to each markdown-list element in the changelog.

Currently, when processing the following example:

```markdown
## [Unreleased]

This release does one thing

### Added

- List with one space
```

The action produces the following output:

```markdown
## [Unreleased]

## [v0.0.9] - 2021-04-03

This release does one thing

### Added

-   List with one space
```

However, the expected output is:

```markdown
## [Unreleased]

## [v0.0.9] - 2021-04-03

This release does one thing

### Added

- List with one space
```

Literally, it adds two extra spaces for each list item. 
It changes (`_` denotes a space):
```
-_List with one space
```
into 
```
-___List with one space
```

I think the list entries should remain unmodified due to the following reasons:

1) https://keepachangelog.com/en/1.0.0/ defines them with a single space
2) Using 3 spaces triggers markdown-lint warning: `MD030/list-marker-space: Spaces after list markers [Expected: 1; Actual: 3] markdownlint (MD030)`

It would be great if this could be fixed 🙂

### PR DESCRIPTION
**Pull request by DervexDev**
Closes #9.

This PR makes changelogs generated by this action 100% accurate with [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) format.

### Before:
```md
...

## [0.2.0] - 2019-09-13

### Added

-   First feature that is gonna make us money.
-   Quite a few bugs, sorry in advance!

### Changed

-   Reworked the login system. You have to provide a password now!

## [0.1.0] - 2019-09-05

### Added

-   Initial release :tada:

[Unreleased]: https://github.com/foo/bar/compare/0.2.0...HEAD

[0.2.0]: https://github.com/foo/bar/compare/0.1.0...0.2.0

[0.1.0]: https://github.com/foo/bar/compare/1625533e04119e8496b14d5e18786f150b4fce4d...0.1.0
```

### After:
```md
...

## [0.2.0] - 2019-09-13

### Added

- First feature that is gonna make us money.
- Quite a few bugs, sorry in advance!

### Changed

- Reworked the login system. You have to provide a password now!

## [0.1.0] - 2019-09-05

### Added

- Initial release :tada:

[unreleased]: https://github.com/foo/bar/compare/0.2.0...HEAD
[0.2.0]: https://github.com/foo/bar/compare/0.1.0...0.2.0
[0.1.0]: https://github.com/foo/bar/compare/1625533e04119e8496b14d5e18786f150b4fce4d...0.1.0
```

I also formatted the code and made some minor improvements.

I would love this change to be released as soon as possible :)

### CODE REVIEW

File: src/updateChangelog.ts [hunk 1]
```diff
@@ -279,7 +275,12 @@ export default async function updateChangelog(
       releaseDate,
       genesisHash,
       owner,
-      repo
+      repo,
+    })
+    .data("settings", {
+      listItemIndent: "one",
+      tightDefinitions: true,
+      bullet: "-",
```
Comment (thomaseizinger) [inline src/updateChangelog.ts:283]: This is the actual fix, right? Can we please split the remaining formatting off into another PR?

**Review by thomaseizinger — COMMENTED** [review]
One request, otherwise LGTM!

I'd appreciate if the re-formatting also comes with CI checks so we don't mess it up again.

File: src/updateChangelog.ts (hunk 1 above)
Comment (DervexDev) [reply src/updateChangelog.ts]: Done. I kept the formatted `CHANGELOG.md`s as they are required for tests.

**Comment by DervexDev** [comment]
> I'd appreciate if the re-formatting also comes with CI checks so we don't mess it up again.

Added CI workflow that run tests.

**Review by thomaseizinger — APPROVED** [review]
Awesome, thanks a lot! :)

**Comment by DervexDev** [comment]
Umm that's weird why does `npm test` fail? I was able to run tests with no problem on my machine.

**Comment by thomaseizinger** [comment]
> Umm that's weird why does `npm test` fail? I was able to run tests with no problem on my machine.

I don't know what the default nodejs version is on GitHub. Might be worth using setup-node to be explicit.

**Comment by DervexDev** [comment]
Done! It was actually webpack's fault.

I also had to ignore one TS error as it actually doesn't affect anything. Tests now work just fine and I was able to build action with no problems!

**Pull request merged**